src/common/decorators/pagination.data.decorator.spec.ts

```diff
@@ -0,0 +1,55 @@
+import { PaginationDataDecorator } from './pagination.data.decorator';
+import { PaginationData } from '../pagination/pagination.data';
+import { Controller, Get, INestApplication, Module } from '@nestjs/common';
+import { fakeCacheService } from '../cache/__tests__/test.cache.module';
+import { Test, TestingModule } from '@nestjs/testing';
+import * as request from 'supertest';
+
+describe('PaginationDataDecorator', () => {
+  let app: INestApplication;
+  let paginationData;
+
+  @Controller()
+  class TestController {
+    @Get()
+    route(@PaginationDataDecorator() data?: PaginationData) {
+      paginationData = data;
```

Comment:
Nitpick: maybe would be better to return the `PaginationData` instead using a class variable? Not a big deal here because there're only 2 simple tests, but I think it's a general best practice, and that will make the route a pure function with no side effects.

Reply:
I do agree but that leaves me with another question: we'd end up testing an HTTP response instead of the actual `PaginationData` object. My goal with these tests was to test if the decorator was working as intended and not to test the result of wrapping pagination data in the body of an HTTP response.

If we return from the route we end up with the following assertions (do notice the `body`):

```typescript
expect(actual.body.limit).toBe(undefined);
expect(actual.body.offset).toBe(undefined);
```

What do you think?